x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/es_deprecations.tsx

```diff
@@ -0,0 +1,212 @@
+                </>
+              )}
+            </span>
+          ),
+          content: (
+            <DeprecationTabContent
+              key="cluster"
+              deprecations={checkupData ? checkupData.cluster : undefined}
+              checkupLabel={i18nTexts.clusterTab.deprecationType}
+              {...commonTabProps}
+            />
+          ),
+        },
+        {
+          id: 'indices',
+          'data-test-subj': 'upgradeAssistantIndicesTab',
+          name: (
+            <span>
+              {i18nTexts.indicesTab.tabName}
+              {checkupData && checkupData.indices.length > 0 && (
+                <>
+                  {' '}
+                  <EuiNotificationBadge>{checkupData.indices.length}</EuiNotificationBadge>
+                </>
+              )}
+            </span>
+          ),
+          content: (
+            <DeprecationTabContent
+              key="indices"
+              deprecations={checkupData ? checkupData.indices : undefined}
+              checkupLabel={i18nTexts.indicesTab.deprecationType}
+              {...commonTabProps}
+            />
+          ),
+        },
+      ];
+    }, [checkupData, error, history, isLoading, resendRequest]);
+
+    useEffect(() => {
+      breadcrumbs.setBreadcrumbs('esDeprecations');
+    }, [breadcrumbs]);
+
+    useEffect(() => {
+      if (isLoading === false) {
+        setTelemetryState(TelemetryState.Running);
+
+        async function sendTelemetryData() {
+          await api.sendTelemetryData({
+            [tabName]: true,
+          });
+          setTelemetryState(TelemetryState.Complete);
+        }
+
+        sendTelemetryData();
+      }
+    }, [api, tabName, isLoading]);
+
+    return (
+      <EuiPageBody>
+        <EuiPageContent>
+          <EuiPageHeader
+            pageTitle={i18nTexts.pageTitle}
+            description={i18nTexts.pageDescription}
+            rightSideItems={[
+              <EuiButtonEmpty
+                href={docLinks.links.upgradeAssistant}
+                target="_blank"
+                iconType="help"
+                data-test-subj="documentationLink"
+              >
+                {i18nTexts.docLinkText}
+              </EuiButtonEmpty>,
+            ]}
+          >
+            <EuiToolTip position="bottom" content={i18nTexts.backupDataButton.tooltipText}>
+              <EuiButton
+                fill
+                href={getUrlForApp('management', {
+                  path: 'data/snapshot_restore',
+                })}
+                iconType="popout"
+                iconSide="right"
+                target="_blank"
+              >
+                {i18nTexts.backupDataButton.label}
+              </EuiButton>
+            </EuiToolTip>
+          </EuiPageHeader>
+
+          <EuiPageContentBody>
+            <EuiTabbedContent
+              data-test-subj={
+                telemetryState === TelemetryState.Running
+                  ? 'upgradeAssistantTelemetryRunning'
```

Comment:
I am not sure I follow why we provide a component state as data-test-subj, do you mind explaining?

Reply:
This was pre-existing behavior. The `data-test-subj` is [used in the functional tests](https://github.com/elastic/kibana/blob/master/x-pack/test/functional/page_objects/upgrade_assistant_page.ts#L74) so I left as-is for now. I imagine I will be making changes to the tests once UA is re-enabled and Kibana deprecations are incorporated in the UI and can revisit it then.